dockstore-webservice/src/main/java/io/dockstore/webservice/resources/proposedGA4GH/ToolsApiExtendedServiceImpl.java

```diff
@@ -391,6 +409,57 @@ public Response setSourceFileMetadata(String type, String id, String versionId,
         throw new CustomWebApplicationException("Could not submit verification information", HttpStatus.SC_BAD_REQUEST);
     }
 
+    @Override
+    public Response submitMetricsData(String id, String versionId, Partner platform, User owner, String description, List<Execution> executions) {
+        // Check that the entry and version exists
+        Entry<?, ?> entry;
+        try {
+            entry = getEntry(id, Optional.of(owner));
+        } catch (UnsupportedEncodingException | IllegalArgumentException e) {
+            return BAD_DECODE_REGISTRY_RESPONSE;
+        }
+
+        Optional<? extends Version<?>> version = getVersion(entry, versionId);
+        if (entry == null) {
```

Comment:
Nit, to me, this check belongs before line 422. If there's no entry, just stop; don't do other stuff then check it if it's null.